nanoFramework.CoreLibrary/System/SpanByte.cs

```diff
@@ -0,0 +1,172 @@
+        /// true if the current span is empty; otherwise, false.
+        /// </summary>
+        public bool IsEmpty => Length == 0;
+
+        /// <summary>
+        /// Copies the contents of this System.Span into a destination System.Span.
+        /// </summary>
+        /// <param name="destination"> The destination System.Span object.</param>
+        /// <exception cref="System.ArgumentException">
+        /// destination is shorter than the source System.Span.
+        /// </exception>
+        public void CopyTo(SpanByte destination)
+        {
+            if (destination.Length < _length - _start)
+            {
+                throw new ArgumentException($"Destination too small");
+            }
+
+            for (int i = 0; i < _array.Length; i++)
+            {
+                destination[i] = _array[i];
+            }
+        }
+
+        /// <summary>
+        /// Forms a slice out of the current span that begins at a specified index.
+        /// </summary>
+        /// <param name="start">The index at which to begin the slice.</param>
+        /// <returns>A span that consists of all elements of the current span from start to the end of the span.</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">start is less than zero or greater than System.Span.Length.</exception>
+        public SpanByte Slice(int start)
+        {
+            if ((start > _length - _start) || (start < 0))
+            {
+                throw new ArgumentOutOfRangeException($"start is less than zero or greater than length");
+            }
+
+            return new SpanByte(_array, _start + start, Length - start - _start);
+        }
+
+        /// <summary>
+        /// Forms a slice out of the current span starting at a specified index for a specified length.
+        /// </summary>
+        /// <param name="start">The index at which to begin this slice.</param>
+        /// <param name="length">The desired length for the slice.</param>
+        /// <returns>A span that consists of length elements from the current span starting at start.</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">start or start + length is less than zero or greater than System.Span.Length.</exception>
+        public SpanByte Slice(int start, int length)
+        {
+            if ((start < 0) || (start + length < 0) || (start + length > _length - _start))
+            {
+                throw new ArgumentOutOfRangeException($"start or start + length is less than zero or greater than length");
+            }
+
+            return new SpanByte(_array, _start + start, length);
+        }
+
+        /// <summary>
+        /// Copies the contents of this span into a new array.
+        /// </summary>
+        /// <returns> An array containing the data in the current span.</returns>
+        public byte[] ToArray()
+        {
+            byte[] array = new byte[Length];
```

Comment:
nit: not sure if nano has this optimization so might be worth to use `_length` directly